### PR TITLE
diskless support root options in addkcmdline 

### PR DESCRIPTION
**Pull request by bybai**
for https://github.ibm.com/DCS-research/WSC-coral/issues/1385#issuecomment-6770314 and https://github.com/xcat2/xcat-core/issues/2470

UT:
1, add addkcmdline for node definition:
```
chdef mid05tor12cn15 addkcmdline='rootflags=rw,mode=755,mpol=interleave:0,8'
```

2,
In debug mode , dracut switch_root:
```
switch_root:/# mount |grep rootfs
rootfs on / type rootfs (rw,size=127184768k,nr_inodes=1987262)
rootfs on /sysroot type tmpfs (rw,relatime,mode=755,mpol=interleave:0,8)
switch_root:/# egrep FilePages /sys/devices/system/node/node{0,8}/meminfo
/sys/devices/system/node/node0/meminfo:Node 0 FilePages:       1049792 kB
/sys/devices/system/node/node8/meminfo:Node 8 FilePages:        778752 kB
--
```
3.
In diskless node:
```
[root@mid05tor12cn15 ~]# mount|grep rootfs
rootfs on / type tmpfs (rw,relatime,mode=755,mpol=interleave:0,8)
[root@mid05tor12cn15 ~]#  egrep FilePages /sys/devices/system/node/node{0,8}/meminfo
/sys/devices/system/node/node0/meminfo:Node 0 FilePages:        727296 kB
/sys/devices/system/node/node8/meminfo:Node 8 FilePages:        703232 kB
```

### CODE REVIEW

File: xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot [hunk 1]
```diff
@@ -77,10 +77,14 @@ if [ -r /rootimg.sfs ]; then
...
+    rootopts="mode=755"
+    if [ -n $rflags ]; then
+        rootopts="$rootopts"",""$rflags"
+    fi
     if [ -z $rootlimit ];then
-        mount -t tmpfs -o mode=755 rootfs $NEWROOT
+        mount -t tmpfs -o $rootopts rootfs $NEWROOT
     else
-        mount -t tmpfs -o mode=755,size=$rootlimit rootfs $NEWROOT
+        mount -t tmpfs -o "$rootopts"size=$rootlimit rootfs $NEWROOT
```
Comment (immarvin) [inline xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot]: mount -t tmpfs -o ${rootopts},size=$rootlimit rootfs $NEWROOT

File: xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot [hunk 2]
```diff
@@ -77,10 +77,14 @@ if [ -r /rootimg.sfs ]; then
 elif [ -r /rootimg.cpio.gz ] || [ -r /rootimg.cpio.xz ]; then
     logger $SYSLOGHOST -t $log_label -p local4.info "Setting up RAM-root tmpfs on downloaded rootimg.cpio.[gz/xz]..."
     echo Setting up RAM-root tmpfs.
+    rootopts="mode=755"
+    if [ -n $rflags ]; then
```
Comment (immarvin) [inline xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot]: `if [ -n "$rflags" ]; then`

File: xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot [hunk 3]
```diff
@@ -77,10 +77,14 @@ if [ -r /rootimg.sfs ]; then
...
+        rootopts="$rootopts"",""$rflags"
+    fi
     if [ -z $rootlimit ];then
```
Comment (immarvin) [inline xCAT-server/share/xcat/netboot/rh/dracut_033/xcatroot]: `     if [ -z "$rootlimit" ];then`

**Comment by immarvin** [comment]
agree to merge

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:31:11 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "[xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:16:11 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:51:07 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:18:49 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:28:03 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:30:49 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:51:17 AM.

This is the only notification you will receive while this person is away.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/3/18
8:28:35 AM.

This is the only notification you will receive while this person is away.

**Comment by robin2008** [comment]
Just comment now it is only for rhel,  need it for ubuntu and sles too when required in future.

**Comment by chuckbrazie** [comment]
I am out of the office until 07/16/2018.

I am out of the office until  Monday July 16.


Note: This is an automated response to your message  "Re: [xcat2/xcat-core]
diskless support root options in addkcmdline (#5366)" sent on 7/4/18
3:06:54 AM.

This is the only notification you will receive while this person is away.

**Pull request merged**